### ISSUE TITLE
"prefer-readonly" incorrectly reported for a private field that is reassigned-to within a nested class

### ISSUE THREAD

**Issue by ashvinsrivatsa** [palantir/tslint#4759]
### Bug Report

- __TSLint version__: 5.17.0
- __TypeScript version__: 3.5.1
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
export class C
{
	private _a = false;

	private readonly _S = class
	{
		private readonly _parent: C;

		constructor(parent: C)
		{
			this._parent = parent;
		}

		doSomething()
		{
			this._parent._a = true;
			console.log(this._parent._a);
		}
	}

	public f()
	{
		(new this._S(this)).doSomething();
	}
}
```

with `tslint.json` configuration:

```json
{ "extends": [ "tslint:all" ] }
```

#### Actual behavior

TSLint reports: 

> Private member variable '_a' is never reassigned; mark it as 'readonly'.

#### Expected behavior

TSLint should not report this error, since the nested class `_S` is permitted to mutate `C`'s private field `_a`.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
